core/src/main/java/hudson/model/UpdateSite.java

```diff
@@ -245,7 +245,16 @@ private FormValidation verifySignature(JSONObject o) throws IOException {
      */
     @Nonnull
     protected JSONSignatureValidator getJsonSignatureValidator() {
-        return new JSONSignatureValidator("update site '"+id+"'");
+        return getJsonSignatureValidator("update site '"+id+"'");
+    }
+
+    /**
+     * Let sub-classes of UpdateSite provide their own signature validator.
+     * @return the signature validator.
+     */
+    @Nonnull
+    protected JSONSignatureValidator getJsonSignatureValidator(String name) {
```

Comment:
cause the signature Validator can also be called by Download Service and in that case the name would be "downloadable" according to the current implementation. I ll fix the deprecation